src/modules/product/controller/productController.ts

```diff
@@ -319,74 +316,74 @@ const buyerAddProductToWishList = async (req: ExtendRequest, res: Response) => {
   }
 };
 const buyerDeleteAllProductFromWishlist = async (
-  req: ExtendRequest,
+  req: any,
   res: Response
 ) => {
   try {
-    await productRepositories.deleteAllWishListByUserId(req.user.id);
+    await productRepositories.deleteAllProductFromWishListById(req.wishList);
     res
       .status(httpStatus.OK)
-      .json({ message: "Your wishlist is cleared successfully." });
-  } catch (error) {
+      .json({ status : httpStatus.OK , message: "Your wishlist is cleared successfully." });
+      } catch (error) {
     res
       .status(httpStatus.INTERNAL_SERVER_ERROR)
       .json({ message: "Internal server error", error: error.message });
   }
 };
 
 const buyerDeleteProductFromWishList = async (
-  req: ExtendRequest,
+  req: any,
   res: Response
 ) => {
   try {
-    await productRepositories.deleteProductFromWishListById(
-      req.user.id,
-      req.params.id
+    await productRepositories.deleteProductFromWishList(
+      req.params.id,
+      req.wishList      
     );
     res
       .status(httpStatus.OK)
-      .json({ message: "The product  removed from wishlist successfully." });
+      .json({ status : httpStatus.OK , message: "The product removed from wishlist successfully." });
   } catch (error) {
     res
       .status(httpStatus.INTERNAL_SERVER_ERROR)
       .json({ message: "Internal server error", error: error.message });
   }
 };
-
-const buyerViewWishLists = async (req: ExtendRequest, res: Response) => {
-  try {
-    const product = await productRepositories.findProductFromWishListByUserId(
-      req.user.id
-    );
-    res.status(httpStatus.OK).json({
-      message: "WishList is fetched successfully.",
-      data: { product },
-    });
-  } catch (error) {
-    res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
-      status: httpStatus.INTERNAL_SERVER_ERROR,
-      error: error.message,
-    });
-  }
-};
-const buyerViewWishList = async (req: ExtendRequest, res: Response) => {
-  try {
-    const product = await productRepositories.findProductfromWishList(
-      req.params.id,
-      req.user.id
-    );
-    res.status(httpStatus.OK).json({
-      message: "WishList is fetched successfully.",
-      data: { product },
-    });
-  } catch (error) {
-    res.status(httpStatus.INTERNAL_SERVER_ERROR).json({
-      status: httpStatus.INTERNAL_SERVER_ERROR,
-      error: error.message,
-    });
-  }
+const buyerViewWishLists = async (req:any,res:Response) => {
+      try{
+        const wishListProducts = await productRepositories.getProductsFromWishlist(req.wishList);
+        const products = wishListProducts.map( (p) => (p as any).products)

```

Comment:
Instead of using `as any` you can create extended type for wishListProducts to include products attributes.